tests/context_tests.rs

```diff
@@ -366,28 +366,26 @@ mod test_pcr_extend_reset {
         // get reset by any D-RTPM events.
         // PCR (TCG PC Client Platform TPM Profile (PTP) for TPM 2.0 Version 1.05 Rev 14)
         let mut context = create_ctx_with_session();
-        let pcr_ses = context.sessions();
+        let pcr_ses = context.sessions().0;
 
         // We start by resetting. We do not place any expectations on the prior contents
-        context.pcr_reset(PcrHandle::Pcr16).unwrap();
+        context.execute_with_session(pcr_ses, |ctx| ctx.pcr_reset(PcrHandle::Pcr16).unwrap());
 
-        // pcr_read is NO_SESSIONS
-        context.clear_sessions();
         // Read PCR contents
         let pcr_selection_list = PcrSelectionListBuilder::new()
             .with_selection(HashingAlgorithm::Sha1, &[PcrSlot::Slot16])
             .with_selection(HashingAlgorithm::Sha256, &[PcrSlot::Slot16])
             .build();
-        let (_, _, pcr_data) = context.pcr_read(&pcr_selection_list).unwrap();
+        // pcr_read is NO_SESSIONS
+        let (_, _, pcr_data) =
+            context.execute_with_session(None, |ctx| ctx.pcr_read(&pcr_selection_list).unwrap());
```

Comment:
Given that some calls are `NO_SESSIONS`, does it make sense to have an `execute_with_no_sessions` method as well?

Reply:
Sure, I had exactly that in my own copy, will put that in here too.